src/apphosting/config.ts

```diff
@@ -36,15 +37,15 @@ export interface Config {
 }
 
 /**
- * Finds the path of apphosting.yaml.
- * Starts with cwd and walks up the tree until apphosting.yaml is found or
+ * Finds the path of a yaml file.
+ * Starts with cwd and walks up the tree until yamlFileName is found or
  * we find the project root (where firebase.json is) or the filesystem root;
  * in these cases, returns null.
  */
-export function yamlPath(cwd: string, fileName: string): string | null {
+export function yamlPath(cwd: string, yamlFileName: string): string | null {
```

Comment:
No it doesn't have to be a yaml file, and I agree that the function should be renamed. Also now that you mention it I don't think this function was intended to search the entire project. Its currently being used by the `apphosting:secrets:*` commands, which i'm guessing so that the command still works when it's run in some subdirectory of the project? i.e `apphosting:secrets:set` should work in project root or in some subdirectory of the project as well.

Reply:
Refactored it a little bit to return the root directory of the apphosting project instead. This makes it easier to be re-used and is more readable.